src/System.Management.Automation/engine/hostifaces/PowerShell.cs

```diff
@@ -4904,6 +4912,8 @@ private void CoreInvoke<TOutput>(IEnumerable input, PSDataCollection<TOutput> ou
             return _invokeAsyncResult;
         }
 
+        // Apartment thread state does not apply to non-Windows platforms.
+#if !UNIX
         /// <summary>
         /// Verifies the settings for ThreadOptions and ApartmentState.
         /// </summary>
```

Comment:
Would it be better to have VerifyThreadSettings just return on Unix rather than all cases that use it?

Reply:
It feels like a wasted call.  `VerifyThreadSettings` is only about apartment threading so I felt it should be completely removed from UNIX builds.